File: Anole/HTMLCenterElement.vb
Namespace Anole

    Public Class HTMLCenterElement : Inherits HTMLElement

        Public Sub New(ByRef document As HTMLDocument, ByVal parent As HTMLElement, ByVal xnode As XmlNode)
            Initialize(document, parent, xnode)
            Me.Alignment = HTMLAlignment.Center
            ParseChildren(xnode)
        End Sub

        Public Overloads Overrides Sub Layout(ByVal renderer As HTMLRenderer, ByVal g As Graphics)
            'If the current band has items, then start a new band.
            If renderer.CurrentBand().Width > 0 Then renderer.NewBand()

            MyBase.Layout(renderer, g)
            renderer.NewBand()
        End Sub

    End Class

End Namespace

File: Anole/HTMLCenterElement.cs
using System;
using System.Xml;
using System.Drawing;

namespace Anole
{
	/// <summary>
	/// Summary description for HTMLCenterElement.
	/// </summary>
	public class HTMLCenterElement : HTMLElement
	{
		public HTMLCenterElement(ref HTMLDocument document, HTMLElement parent, XmlNode xnode)
		{
			Initialize(ref document, parent, xnode);
			this.Alignment=HTMLAlignment.Center;
			ParseChildren(xnode);
		}

		public override void Layout(HTMLRenderer renderer, Graphics g)
		{
			//If the current band has items,
			//then start a new band.
			if(renderer.CurrentBand().Width>0) renderer.NewBand();

			base.Layout(renderer,g);
			renderer.NewBand();
		}
	}
}
